operator/src/main/resources/scripts/introspectDomain.sh

```diff
@@ -142,58 +189,78 @@ if [ ${DOMAIN_SOURCE_TYPE} == "FromModel" ]; then
     createWLDomain || exit 1
     created_domain=$DOMAIN_CREATED
     trace "Create domain return code = " ${created_domain}
-else
+  else
     created_domain=1
-fi
-
-traceTiming "INTROSPECTOR '${DOMAIN_UID}' MII CREATE DOMAIN END" 
-
+  fi
 
-# check DOMAIN_HOME for a config/config.xml, reset DOMAIN_HOME if needed
+  traceTiming "INTROSPECTOR '${DOMAIN_UID}' MII CREATE DOMAIN END" 
 
-exportEffectiveDomainHome || exit 1
+  # check DOMAIN_HOME for a config/config.xml, reset DOMAIN_HOME if needed
 
-# list potentially interesting env-vars and dirs after they're updated by export.*Homes
+  exportEffectiveDomainHome || exit 1
 
-traceEnv after
-traceDirs after
+  # list potentially interesting env-vars and dirs after they're updated by export.*Homes
 
-# check if we're using a supported WebLogic version
-# (the check  will log a message if it fails)
+  traceEnv after
+  traceDirs after DOMAIN_HOME DATA_HOME
 
-checkWebLogicVersion || exit 1
+  # check if we're using a supported WebLogic version
+  # (the check  will log a message if it fails)
 
-# start node manager
-# run instrospector wlst script
-if [ ${created_domain} -ne 0 ]; then
+  checkWebLogicVersion || exit 1
 
+  # start node manager
+  # run instrospector wlst script
+  if [ ${created_domain} -ne 0 ]; then
     traceTiming "INTROSPECTOR '${DOMAIN_UID}' MII NM START" 
 
-    # start node manager -why ??
+    # introspectDomain.py uses an NM to setup credentials for the server NMs
+    #  (see 'nmConnect' in introspectDomain.py)
     trace "Starting node manager"
     ${SCRIPTPATH}/startNodeManager.sh || exit 1
 
     traceTiming "INTROSPECTOR '${DOMAIN_UID}' MII NM END" 
-
     traceTiming "INTROSPECTOR '${DOMAIN_UID}' MII MD5 START"
-
     traceTiming "INTROSPECTOR '${DOMAIN_UID}' MII NM END" 
-
     traceTiming "INTROSPECTOR '${DOMAIN_UID}' MII MD5 START"
 
     # put domain secret's md5 cksum in file '/tmp/DomainSecret.md5'
     # the introspector wlst script and WL server pods will use this value
     generateDomainSecretMD5File '/tmp/DomainSecret.md5' || exit 1
 
     traceTiming "INTROSPECTOR '${DOMAIN_UID}' MII MD5 END"
-
     traceTiming "INTROSPECTOR '${DOMAIN_UID}' INTROSPECT START"
 
     trace "Running introspector WLST script ${SCRIPTPATH}/introspectDomain.py"
     ${SCRIPTPATH}/wlst.sh ${SCRIPTPATH}/introspectDomain.py || exit 1
 
     traceTiming "INTROSPECTOR '${DOMAIN_UID}' INTROSPECT END"
-fi
-trace "Domain introspection complete"
+  fi
+  trace "Domain introspection complete"
+}
 
-exit 0
+# we have different log file modes in case we need to revert 'tee' mode
+
+case "${INTROSPECTOR_LOG_FILE_MODE:-tee}" in
```

Comment:
Implementation note: There are no plans to document INTROSPECTOR_LOG_FILE_MODE - default is 'tee'.